### PR TITLE
Fixes a revenant runtime

### PR DESCRIPTION
**Pull request by Bjamcham**
## What Does This PR Do
Adds a check to prevent the ghost gas analyzer from creating runtimes with revenants

## Why It's Good For The Game
Bugs are bad. [NO GBP Update]

## Testing
Spawned in as a revenant and ghost, checked that it could be used and did not runtime.

## Changelog
NPFC

### CODE REVIEW

File: code/_onclick/observer_onclick.dm [hunk 1]
```diff
@@ -93,6 +93,8 @@
 		user.forceMove(get_turf(target))
 
 /obj/machinery/atmospherics/attack_ghost(mob/dead/observer/user)
+	if(!istype(user)) // Make sure user is actually an observer. Revenents also use attack_ghost, but do not have the toggle gas analyzer var.
```
Comment (Burzah) [inline code/_onclick/observer_onclick.dm:96]:
```suggestion
	if(!isobserver(user)) // Make sure user is actually an observer. Revenents also use attack_ghost, but do not have the toggle gas analyzer var.
```
Would using isobserver be better for this case?

Comment (GDNgit) [reply code/_onclick/observer_onclick.dm:96]: no

**Review by GDNgit — APPROVED** [review]
your gbp will be updated do not resist

File: code/_onclick/observer_onclick.dm (hunk 1 above)
Comment (Burzah) [reply code/_onclick/observer_onclick.dm:96]: Disregard. I see the issue now.

**Review by Burzah — APPROVED** [review]

**Review by lewcc — APPROVED** [review]

**Pull request merged**